operators/pkg/controller/elasticsearch/driver/downscale_test.go

```diff
@@ -0,0 +1,709 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package driver
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/elastic/cloud-on-k8s/operators/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/reconciler"
+	esclient "github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/nodespec"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/observer"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/reconcile"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/sset"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Sample StatefulSets to use in tests
+var (
+	sset3Replicas = nodespec.CreateTestSset("sset3Replicas", "7.2.0", 3, true, true)
+	sset4Replicas = nodespec.CreateTestSset("sset4Replicas", "7.2.0", 4, true, true)
+)
+
+// fakeESClient mocks the ES client to register function calls that were made.
+type fakeESClient struct { //nolint:maligned
```

Comment:
This works, but should we be using gomock instead? It seems like it would be a little more idiomatic, but this code is pretty small (right now) so I don't feel too strongly about it.

Reply:
Yes, I did hesitate but I kind of like the current minimalistic approach without any extra mock library required :)